### ISSUE TITLE
Some log files are empty (BB #1108)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#1108]
This issue was migrated from bitbucket.
**Original Reporter:** PadreWoW
**Original Date:** 16.08.2015 16:37:48 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108

<hr>

chat.log, characters.log, gm_commands.log - files are empty

They are present in conf but they are empty

Other log files are not full:

I see something like this in server.log for example:

**2015-08-16 19:31:00 WORLD: Starting weather system for zon** <- (last string)

and this thing in db_errors.log:

2015-08-16 19:30:32 SourceEntry 37434 in `condi <- (last string)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 16:38:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20950047

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 16:50:33 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20950157

<hr>

Last rev WITHOUT any third-party patches

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Dikkedeur
**Original Date:** 16.08.2015 18:26:33 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20951024

<hr>

Padrewow > https://forums.oregon-core.net/viewtopic.php?id=287
but what i found strange is that my system.err is full of crap 

```
#!c++

^[[01;31m[Err] ^[[00;31mCastSpell: unknown spell id 0 by caster: creature (Entry: 15483)^[[0m
^[[01;31m[Err] ^[[00;31mCastSpell: unknown spell id 0 by caster: creature (Entry: 15483)^[[0m
CASTED 1804 on GUID: 17370385831902067474, entry: 123330
CASTED 3365 on GUID: 17370386845866680336, entry: 183767
CASTED 1804 on GUID: 17370385831902067473, entry: 123330
^[[01;31m[Err] ^[[00;31mCastSpell: unknown spell id 0 by caster: creature (Entry: 7414)^[[0m
^[[01;31m[Err] ^[[00;31mCastSpell: unknown spell id 0 by caster: creature (Entry: 7414)^[[0m
^[[01;31m[Err] ^[[00;31mCastSpell: unknown spell id 0 by caster: creature (Entry: 7414)^[[0m
CASTED 2575 on GUID: 17370383791809369425, entry: 1731
CASTED 2575 on GUID: 17370383791809369425, entry: 1731
CASTED 1804 on GUID: 17370385831902067474, entry: 123330
CASTED 2575 on GUID: 17370383791809369425, entry: 1731
CASTED 2575 on GUID: 17370383791809399465, entry: 1731
```

and not getting overwritten after core restart.. so it gets big in rather fast time.

same with Server.log (although its with date ) it can get rather big in short time

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 18:48:06 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20951187

<hr>

```
#!c++

PidFile = ""
LogMask = 4083
LogFile = "D:\Xampp\htdocs\ruleforweb\logs\server.log"
ChatLogFile = "D:\Xampp\htdocs\ruleforweb\logs\chat.log"
LogTimestamp = 0
WorldLogFile = ""
DBErrorLogFile = "D:\Xampp\htdocs\ruleforweb\logs\db_errors.log"
CharLogFile = "D:\Xampp\htdocs\ruleforweb\logs\characters.log"
CharLogTimestamp = 0
GmLogFile = "D:\Xampp\htdocs\ruleforweb\logs\gm_commands.log"
GmLogTimestamp = 0
GmLogPerAccount = 0
RaLogFile = ""
ArenaLogFile = "D:\Xampp\htdocs\ruleforweb\logs\arena.log"
ArenaLogExtendedInfo = 1
LogColors = "14 11 4 3 1 1 2 7 5 0 4 0 1 3 2 4 0"
DBLogMask = 0
LogSQLFilename = "D:\Xampp\htdocs\ruleforweb\logs\logSQL.sql"
LogSQLTimestamp = 1
ChatLogs.Channel      = 1
ChatLogs.SysChan      = 1
ChatLogs.Whisper      = 1
ChatLogs.Party        = 1
ChatLogs.Raid         = 1
ChatLogs.Guild        = 1
ChatLogs.Public       = 1
ChatLogs.Addon        = 0
ChatLogs.BattleGround = 1
ChatLogTimestamp = 0
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 18:48:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20951192

<hr>

Files are there but they are empty

LogMask = 4083 and nothing happens

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 18:53:14 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20951220

<hr>

(STRING = 1 + BASIC = 2 + ERROR = 16 + ERROR_DB = 32 + SQL = 64 + ARENA = 128 + WARDEN = 256 + CHAT = 512 + COMMAND = 1024 + CHAR = 2048) = 4083

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 19:03:04 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20951309

<hr>

I can see log information on screen but not in files

![Untitled.png](https://bitbucket.org/repo/kMeMB/images/4250004402-Untitled.png)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** HEGladios
**Original Date:** 16.08.2015 20:29:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20952160

<hr>

Ehh all flags combined should be 4095, not 4083 :P

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** desteny
**Original Date:** 16.08.2015 20:47:30 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20952305

<hr>

yes, 4083  is all without DEBUG and DETAIL

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 16.08.2015 20:59:10 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20952380

<hr>

Again - I CAN SEE LOGS ON DISPLAY!!!!!

But files are empty!!!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 18.08.2015 09:53:02 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-20998127

<hr>

any news? this is really big problem!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 21.08.2015 13:49:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21084744

<hr>

any news? this is really big problem!

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 24.08.2015 09:04:01 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21185181

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 25.08.2015 01:21:11 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21212935

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 11:53:52 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21461273

<hr>

Zaffy can you fix it?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Dikkedeur
**Original Date:** 02.09.2015 12:33:07 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21462161

<hr>

i dont have those empty log files, so i think that you should try different bitmask padrewow, since thats where the problem might be..
i dont know what you want to log, but i dont log that much since that affects performance as well and also gives loads of files and crap :P

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 13:10:08 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21462942

<hr>

Other log files are not full:

I see something like this in server.log for example:

2015-08-16 19:31:00 WORLD: Starting weather system for zon <- (last string)

and this thing in db_errors.log:

2015-08-16 19:30:32 SourceEntry 37434 in `condi <- (last string)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 14:05:30 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21464257

<hr>

Can I see your config file and your server.log file?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 14:24:54 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21464681

<hr>

This is MY Server.log with LogMask = 115 and LogFile = "Server.log" (last rev clear core without any third-party scripts)

On line 136, you can see exactly what I mean (136 is the LAST line)

![serverlog.png](https://bitbucket.org/repo/kMeMB/images/7550563-serverlog.png)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 14:35:20 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21465005

<hr>

LogMask = 3699 - Default + GM Commands + Chat + Char

gm_commands.log - not all lines just like server.log

characters.log - empty

chat.log - empty

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 02.09.2015 14:41:13 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21465169

<hr>

DB errors are missing when core already started

Example:

there are DB errors with column ONLINE but there are no such lines in db_errors.log
![dberror.png](https://bitbucket.org/repo/kMeMB/images/162217525-dberror.png)

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Dikkedeur
**Original Date:** 02.09.2015 19:14:47 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21470089

<hr>

LogMask = 3331

LogFile = "Server.log"

ChatLogFile = "" 

LogTimestamp = 1

WorldLogFile = "world.log"

DBErrorLogFile = "db_errors.log"

CharLogFile = "characters.log"

CharLogTimestamp = 1

GmLogFile = "gm.log"

GmLogTimestamp = 1

GmLogPerAccount = 1

GmLogTimestamp = 1

GmLogPerAccount = 1

RaLogFile = "ra_commands.log"

ArenaLogFile = "arena.log"

ArenaLogExtendedInfo = 1

LogColors = "0 6 4 3 1 1 2 7 5 0 4 0 1 3 2 4 0"

DBLogMask = 0

ChatLogs.Channel      = 0

ChatLogs.SysChan      = 0

ChatLogs.Whisper      = 0

ChatLogs.Party        = 0

ChatLogs.Raid         = 0

ChatLogs.Guild        = 0

ChatLogs.Public       = 0

ChatLogs.Addon        = 0

ChatLogs.BattleGround = 0

ChatLogTimestamp = 0

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 03.09.2015 06:20:17 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21479460

<hr>

and you have no problems with log files?

all files are full, and information in files updates dynamically?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 03.09.2015 15:17:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21494550

<hr>

I can definitely confirm it on Windows.

@Dikke Duer Which OS are you using?

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Dikkedeur
**Original Date:** 03.09.2015 16:16:48 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21496170

<hr>

i use ubuntu server 14.0.2 LtS Lord :P

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.09.2015 14:57:16 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-21678076

<hr>

Any news? :(((

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** PadreWoW
**Original Date:** 10.10.2015 21:23:12 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-22403597

<hr>

RIP oregon :(

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Zaffy
**Original Date:** 11.10.2015 19:10:40 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/1108/#comment-22413350

<hr>

Fix #1108 - issue with log buffers not being flushed.

→ <<cset e387a13fb786>>
